src/GitVersion.Core/Git/IReferenceCollection.cs

```diff
@@ -7,7 +8,7 @@ public interface IReferenceCollection : IEnumerable<IReference>
         IReference Head { get; }
         IReference this[string name] { get; }
         void Add(string name, string canonicalRefNameOrObjectish, bool allowOverwrite = false);
-        void UpdateTarget(IReference directRef, IObjectId targetId);
+        void UpdateTarget(IReference directRef, IObjectId targetId, ILog log);
```

Comment:
I would prefer not to pass the `log`, maybe the consumer of this method be _wrapped_ by the `OperationWithExponentialBackoff`

Reply:
Agreed, that's not pretty. The problem is that the LockedFileException is defined by LibGit2Sharp. I worked around it by wrapping it in another Exception defined as part of the IReferenceCollection abstraction. Also not very pretty, but architecturally cleaner. https://github.com/GitTools/GitVersion/pull/2581/commits/3902675beb810bfdb80f9d2e256901a1c1b36b63